packages/valory/skills/transaction_settlement_abci/rounds.py

```diff
@@ -379,17 +362,24 @@ class ValidateTransactionRound(VotingRound):
     def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
         """Process the end of the block."""
         # if reached participant threshold, set the result
+
         if self.positive_vote_threshold_reached:
+            # We obtain the latest tx hash from the `tx_hashes_history`.
+            # We keep the hashes sorted by their finalization time.
+            # If this property is accessed before the finalization succeeds,
+            # then it is incorrectly used.
+            synchronized_data = cast(SynchronizedData, self.synchronized_data)
+            if not synchronized_data.to_be_validated_tx_hash:
+                return synchronized_data, Event.INCORRECT_SERIALIZATION
```

Comment:
we now return a incorrect serialization event instead of raising an Error

Reply:
If we don't have a tx hash than something went wrong during serialization. The best alternative seems `FINALIZATION_FAILED` which I think is less appropriate.